BUILDING.md

```diff
@@ -425,3 +425,13 @@ To make `./myModule.js` available via `require('myModule')` and
 ```console
 > .\vcbuild link-module './myModule.js' link-module './myModule2.js'
 ```
+
+## Note for downstream distributors of Node.js
+
+The Node.js ecosystem is reliant on ABI compatibility within a major
+release. To maintain ABI compatibility it is expected that production
+builds of Node.js will be built against the same version of dependencies as the
+project vendors. If Node.js is to be built against a different version of a
+dependency please create a custom `NODE_MODULE_VERSION` to ensure ecosystem
```

Comment:
I think we should flesh out a bit more what is involved in a custom `NODE_MODULE_VERSION`. The problem is that we don't necessarily have "room" in the incrementally increasing versions to anticipate needs apriori. 

My proposal is:
* Reserve 8-bit MSB in the `NODE_MODULE_VERSION` for 'distribution'.
* Official builds would use 0. Electron, and other distributors could use a different value if the binary they are shipping is not ABI compatible.

Reply:
reserving a number of bits makes sense to me.

Reply:
+1. Does someone want to push a commit that adds this specific documentation?

Reply:
@ofrobots would you be able to supply explicit copy for this? Otherwise I'd like to land this in an iteration of this PR